Cloud-Native Rework, Database Flexibilization and Encryption, Bot Support

henry701 opened the pull request:
These commits effectively make the application database-agnostic (using SQLAlchemy) and cloud-native (tested on Heroku) by receiving more configuration as environment variables.

There are also improvements from another fork that adds bot support (from @kilianovski).

To top it off, database encryption is optionally supported (although it uses base64 which increases storage quite a bit) so you can use an untrusted database as a datastore!

If merging this, you may want to add your details to .github/funding too, in the very low-probability but still possible case that someone has financial interest in the project.

henry701 committed:
docs: Improve readme grammar

henry701 committed:
refactor: Copy only necessary files for Dockerfile to function

henry701 committed:
docs: Use - instead of '–' for values

henry701 committed:
chore: Add hidden files/folders to gitignore

henry701 committed:
chore: Update dependencies

henry701 committed:
feat!: Major rework

feat: Use SQLAlchemy to store both session and application data
feat: Bot integration courtesy of kilianovski fork
refactor: Separation of folders based on concern and usage
refactor: A better-layered DOCKERFILE and docker-compose

henry701 committed:
refactor: Add names to docker-compose

henry701 committed:
docs: Update Readme

henry701 committed:
docs: Update .env.example

henry701 committed:
chore: Add cryptg dependency (prepare to download media)

henry701 committed:
chore: minor rework
refactor: Fix log level
feat: Store peer_id properly, update deleted boolean
fix: Fix misc issues
refactor: Move some code around, remove other useless code

henry701 committed:
fix: Fix cryptg dependency

henry701 committed:
feat: Attempt to implement encryption

henry701 committed:
fix: Fix dependency and monkey-patch

henry701 committed:
chore: Add certificates to .gitignore just in case

henry701 committed:
fix!: Actually fix and test cryptographic SQLAlchemy storage

henry701 committed:
feat!: Make app possible to run with PostgreSQL backend
fix!: Fix invocation of format_default_message_text
refactor!: Change which fields are encrypted and which are not

henry701 committed:
fix!: There is no get_event_sender for on_message_deleted event

henry701 committed:
fix!: Fallback logics for getting entities because Telethon bad

henry701 committed:
fix: Don't set SQLAlchemy level

henry701 committed:
fix!: Pass client together with message for notification of deletion
refactor: Add some relatively critical TODOs

henry701 committed:
refactor!: Rework database (again) to properly store message source
chore!: Handle signals to gracefully close application
feat!: Add more logging granularity
fix!: Fix some issues with Telethon being unable to find peers

henry701 committed:
refactor!: Flexibilize supports using dockerfile ARGs

henry701 committed:
fix!: Fix database issues and self peer special case

henry701 committed:
fix:Turn buggy sqreen off by default

henry701 committed:
fix: Correctly handle error while importing sqreen

henry701 committed:
Grammar fixes to README.md

henry701 committed:
Add Github Funding

henry701 committed:
Update funding with paypal link

henry701 committed:
feat: Add USE_HTTPS and USE_EXTERNAL_PORT environment variables for hosting flexibilization (Heroku)
fix: Fix environment variable conversion to bool

henry701 committed:
fix: Attempt to fix query in case chat_id and/or peer_type is set for deleted message

henry701 committed:
chore: Update .gitignore with .gitignore exception

henry701 committed:
fix: Attempt to fix join again

ValeriyMaslenikov commented:
Hey, @henry701. I've found that you made a really good job to improve this project. I'm pretty busy right now, so I'm going to take some time to review your Pull Request. 

Thanks for what you have done!

henry701 committed:
Once again attempt to fix query when a message is deleted inside a group, using has method this time

henry701 committed:
chore: Add more logging information for missing deleted message, including compiled query

henry701 committed:
fix: Attempt to fix query for when chat_id is present, with OR clause

henry701 committed:
fix: Add handling for duplicated auth key error on connect

henry701 committed:
fix: Also auto-relog the bot too

henry701 committed:
fix: Reorder the client attribution order for the last fix, unset client on aexit.

henry701 committed:
fix: Move re-auth logic inside bot

henry701 committed:
fix: Add initial sleep seconds period to prevent duplicate usage (kludge)

henry701 committed:
fix: zzz python convert to int

henry701 committed:
fix: Delete session forceps instead of log_out because log_out implies connect

henry701 committed:
fix: Recreate client after deleting session with duplicate auth key

henry701 committed:
fix: Overwrite session after delete

henry701 committed:
feat: Notify about unknown deleted messages with the chat peer where they came from, their IDs, and the number of unknowns, at least.

henry701 committed:
fix: Fix typing

henry701 committed:
fix!: Use exclusion for unloaded messages, pass all instead of looping the IDs

henry701 committed:
fix!: Fix missing await on format_default_unknown_message_text

henry701 committed:
chore: Update Telethon

henry701 committed:
fix!: Handle missing input_chat appropriately, unfortunately

henry701 committed:
fix: Properly format message

henry701 committed:
fix: Fix cache_ok warning for updated SQLAlchemy, and pin SQLAlchemy version (too central to be let flexible)

henry701 committed:
fix!: Fix "or" clause inside query

henry701 committed:
fix: Fix encrypted chat to_input_type case

henry701 committed:
fix!: Make get_mention_username more robust and generic (and rename it to get_mention_text)

henry701 committed:
chore: Remove the useless `or` with from_peer_id

henry701 committed:
chore: Add literal_binds true flag to query print warning method

henry701 committed:
Commit .dockerignore

henry701 committed:
fix: Resolve marked peer ID, finally fixing the query issue for deleted messages on chats. God, Telegram and Telethon APIs are a mess.

henry701 committed:
fix!: Ignore peer type after calling resolve_id, because it returns a tuple

henry701 committed:
fix: Test for None before using resolve_id

henry701 committed:
fix: Anonymous tuple for proper unpacking...

henry701 committed:
fix: Add value to unpack expression

henry701 committed:
feat!: Use Gunicorn + Flask for web and background thread for asyncio
refactor!: Further refactor and separate app methods

henry701 committed:
feat: New endpoints for better control
refactor: Use sessionmaker begin method instead of manual commit
chore: Cleanup old logs etc

henry701 committed:
chore: Tidy up .env.example

henry701 committed:
feat: Add sample fly.toml file

henry701 committed:
fix: Use core mode for SQLAlchemy Session Container so that session expiration due to disconnection does not break the application (happens on fly.io, could happen on other clouds with stingy firewalls)

nsauk commented on .github/FUNDING.yml:
```diff
@@ -0,0 +1,6 @@
+  # - henry701
+
+custom:
+  - https://www.paypal.com/donate?hosted_button_id=TYWQ22HUDANXU
```
:roll_eyes:

henry701 committed:
fix: Fix future count

henry701 committed:
chore: Update image version

henry701 committed:
feat: Add health check endpoint

henry701 committed:
fix: Typing fixes, also await refresh_client

henry701 committed:
fix: Fix typing hints

henry701 committed:
docs: Add roadmap item

henry701 committed:
chore: Add pooling and pre_ping parameters to SQLAlchemy Engine

henry701 committed:
fix: Do not fully commit (and thus do not close) session when adding new peers, just flush it

henry701 committed:
chore: Upgrade some log levels

henry701 committed:
chore: Add logging for errors, to figure out why cleanupper job is not working as intended (only runs once)

henry701 committed:
fix: Fix stop_event behavior

henry701 committed:
chore: Add missing assert

henry701 committed:
fix: Add missing 'await' for self recursive call

henry701 committed:
fix: Use awaitables instead of python futures for asyncio gather call

Commit:
Postgresql fixes

henry701 committed:
Merge branch 'master' into 'master'

Postgresql fixes

See merge request henry701/tg-notify-deleted-messages!1

henry701 committed:
fix: Fix auth error messages and 2FA flow

henry701 committed:
docs: Tidy up readme, clarify roadmap

henry701 committed:
feat: Add feature to ignore channels and groups. That'll surely help with the spam from channel comments...

henry701 committed:
feat: Flesh out the ignore flags
fix: Fix some issues with input entities by using full entities instead
fix: Fix typing hints in general

henry701 committed:
fix: Remove extraneous logs (whoops)

henry701 committed:
fix: Upgrade telethon version, do not ignore [mega/super]groups by default, comment about using a false join_to_send attribute to check for discussion groups

henry701 committed:
feat: As a workaround for being unable to see easily if a group is a discussion group of a channel, set a threshold to ignore messages from groups over this amount of member/participants.

henry701 committed:
fix: Use join_to_send when present, move expensive member ignore threshold validation to be the last validation

henry701 committed:
feat: Filter by chat before filtering by message on filter_deleted_messages_for_event, where possible

henry701 committed:
fix: Move the chat filter-all validation up a level so that it also works for ignoring notification of unknown deleted messages

henry701 committed:
fix: Log "(no query)" if query is None

henry701 committed:
fix: Add missing await

henry701 committed:
perf: Early exit when deleted_ids is empty on MessageDeleted event

henry701 committed:
Use exec variant, `as` -> `AS`

henry701 committed:
Improve app logging, remove sqreen support, fix sqlite start

henry701 committed:
use trace log for on_new_message logs, evaluate own message deletion parameter on message deleted and not on message received, add message preloading

henry701 committed:
Fix message preloading and attachment downloading, update dependencies, small refactor

henry701 committed:
Add more Telegram checks to the health check

henry701 committed:
Add retry to database interaction logic using tenacity lib

henry701 committed:
Small fix await preload_message (oops)

henry701 committed:
Add uwsgi support, add logout endpoint, improve logging, preload dialogs in parallel

henry701 committed:
Update fly example file

henry701 committed:
Increase pooling

henry701 committed:
Introduce max parallelism, increase timeouts

henry701 committed:
Add retry to health check

henry701 committed:
Fix store_message lazy peer logic using separate transactions and retries

henry701 committed:
Don't assume we have every message in the database: Always preload and check. Create concurrency env variables for config. Add event handler for new messages and preload the messages in parallel.

henry701 committed:
Session.add -> Session.merge [Upsert Everything]

henry701 committed:
Fix graceful exit and a ton of other small bugs

henry701 committed:
Improve preloading status reporting and graceful exit messaging

henry701 committed:
Optimize preloading to skip by chat. Small refactor for ignore logic

henry701 committed:
Add HTTP authentication based on secret

henry701 committed:
Exit on logout, change some info loggers to debug

henry701 committed:
Fix tasklen logging

henry701 committed:
Configurable preload status interval

henry701 committed:
Optimization to not re-check message chat ignore logic repeteadly when preloading messages, small refactor for loading logic

henry701 committed:
Clarify preloading finished log

henry701 committed:
Fix swapped args

henry701 committed:
PyPy migration in progress

henry701 committed:
Finish PyPy conversion

henry701 committed:
Optimize Dockerfile

henry701 committed:
Add pg8000 support since psycopg2 does not work with PyPy

henry701 committed:
Change log levels to debug

henry701 committed:
Add hypercorn and nginx support. Use postgresql in dockerfile for testing. Patch a bug in sqlalchemy session schema. Fix a bug in our own schema. And more fixes and other adjustments than I could count.

henry701 committed:
Adjust example envs

henry701 committed:
Add a crucial missing arg for non-ssl intra-dc connection :)

henry701 committed:
even more conservative params

henry701 committed:
dos2unix fixes

henry701 committed:
DB fixes

henry701 committed:
Only load one message at a time on iter_messages

henry701 committed:
Adjust config

henry701 committed:
more lenient configs for reverse proxy

henry701 committed:
Revert back to CPython but keep support for PyPy via a huge Dockerfile refactor

henry701 committed:
Make _MAX_CHUNK_SIZE override conditional

henry701 committed:
duh

henry701 committed:
Adjust example env

henry701 committed:
Upgrade to SQLAlchemy 2 and PsycoPG3

henry701 committed:
add libpq for both psycopgs if any enabled

henry701 committed:
jesus christ the media size conditional was inverted

henry701 committed:
tweak params

henry701 committed:
Increase file DL limit to 10mb
